diff --git a/recipes-extended/entservices/entservices-appgateway.bb b/recipes-extended/entservices/entservices-appgateway.bb
index 35991dc1..eedf4add 100644
--- a/recipes-extended/entservices/entservices-appgateway.bb
+++ b/recipes-extended/entservices/entservices-appgateway.bb
@@ -2,9 +2,9 @@ SUMMARY = "ENTServices AppGateway plugins"
 LICENSE = "Apache-2.0"
 LIC_FILES_CHKSUM = "file://LICENSE;md5=9adde9d5cb6e9c095d3e3abf0e9500f1"
 
-PV = "0.2.0.0"
+PV = "0.3.1.0"
 PR = "r0"
-SRCREV = "41b3ed691a11435690d985816e7d617d3369f20c"
+SRCREV = "6c0b5f93f904a07d5ff0bd5a54feca7f18fe0570"
 
 S = "${WORKDIR}/git"
 inherit cmake pkgconfig
@@ -14,6 +14,10 @@ SRC_URI = "${CMF_GITHUB_ROOT}/entservices-appgateway;${CMF_GITHUB_SRC_URI_SUFFIX
 PACKAGE_ARCH = "${MIDDLEWARE_ARCH}"
 TOOLCHAIN = "gcc"
 
+#Announce as firebolt provider
+PROVIDES = "virtual/firebolt"
+RPROVIDES:${PN} = "virtual/firebolt"
+
 DEPENDS += "wpeframework wpeframework-tools-native wpeframework-clientlibraries entservices-apis"
 RDEPENDS:${PN} += "wpeframework entservices-apis"
 
diff --git a/recipes-extended/entservices/entservices-rdkappmanagers.bb b/recipes-extended/entservices/entservices-rdkappmanagers.bb
index 01c28232..3b26c125 100644
--- a/recipes-extended/entservices/entservices-rdkappmanagers.bb
+++ b/recipes-extended/entservices/entservices-rdkappmanagers.bb
@@ -8,9 +8,10 @@ PR ?= "r0"
 S = "${WORKDIR}/git"
 inherit cmake pkgconfig
 
-SRCREV = "cdc08633bc299c509dc51fc2f2fc59b865e5906b"
+SRCREV = "63b99e7bba8e48af63e44b7ac0f3357ecf928d8f"
 
 SRC_URI = "${CMF_GITHUB_ROOT}/entservices-appmanagers;${CMF_GITHUB_SRC_URI_SUFFIX}"
+#SRC_URI += "file://0100-Update-pacakgeType.patch"
 
 PACKAGE_ARCH = "${MIDDLEWARE_ARCH}"
 TOOLCHAIN = "gcc"
@@ -56,6 +57,7 @@ PACKAGECONFIG ?= " telemetrysupport \
     ${@bb.utils.contains('DISTRO_FEATURES', 'DAC-sec',              'ocicontainersec', '', d)} \
     ${@bb.utils.contains('DISTRO_FEATURES', 'opencdm', 'opencdmi', '', d)} \
     ${@bb.utils.contains('DISTRO_FEATURES', 'rialto_in_dac', 'rialtodac', '', d)} \
+    ${@bb.utils.contains('DISTRO_FEATURES', 'enable_ralf', ' ralfsupport', '', d)} \
 "
 
 inherit features_check
@@ -73,13 +75,14 @@ PACKAGECONFIG[rdknativescript]      = "-DPLUGIN_NATIVEJS=ON,-DPLUGIN_NATIVEJS=OF
 PACKAGECONFIG[opencdmi]             = "-DPLUGIN_OPENCDMI=ON"
 PACKAGECONFIG[telemetrysupport]     = "-DBUILD_ENABLE_TELEMETRY_LOGGING=ON,,telemetry,telemetry"
 PACKAGECONFIG[telemetry]            = "-DPLUGIN_TELEMETRY=ON,,iarmbus iarmmgrs entservices-apis rfc rbus,iarmbus entservices-apis rfc rbus"
-PACKAGECONFIG[runtimemanager]       = "-DPLUGIN_RUNTIME_MANAGER=ON ${RUNTIMEMANAGER_PLUGIN_ARGS},-DPLUGIN_RUNTIME_MANAGER=OFF,entservices-apis,entservices-apis"
+PACKAGECONFIG[runtimemanager]       = "-DPLUGIN_RUNTIME_MANAGER=ON ${RUNTIMEMANAGER_PLUGIN_ARGS},-DPLUGIN_RUNTIME_MANAGER=OFF,entservices-apis iptables,entservices-apis iptables"
 PACKAGECONFIG[packagemanager]       = "-DPLUGIN_PACKAGE_MANAGER=ON ${PACKAGEMANAGER_PLUGIN_ARGS} -DLIB_PACKAGE=ON -DSYSROOT_PATH=${STAGING_DIR_TARGET},-DPLUGIN_PACKAGE_MANAGER=OFF -DLIB_PACKAGE=OFF,curl virtual/libpackage,curl virtual/libpackage"
 PACKAGECONFIG[lifecyclemanager]     = "-DPLUGIN_LIFECYCLE_MANAGER=ON,-DPLUGIN_LIFECYCLE_MANAGER=OFF,websocketpp entservices-apis,entservices-apis"
 PACKAGECONFIG[storagemanager]       = "-DPLUGIN_STORAGE_MANAGER=ON,-DPLUGIN_STORAGE_MANAGER=OFF,entservices-apis,entservices-apis"
 PACKAGECONFIG[appmanager]           = "-DPLUGIN_APPMANAGER=ON,-DPLUGIN_APPMANAGER=OFF,entservices-apis,entservices-apis"
 PACKAGECONFIG[preinstallmanager]    = "-DPLUGIN_PREINSTALL_MANAGER=ON,-DPLUGIN_PREINSTALL_MANAGER=OFF,entservices-apis,entservices-apis"
 PACKAGECONFIG[downloadmanager]      = "-DPLUGIN_DOWNLOADMANAGER=ON -DLIB_PACKAGE=ON -DSYSROOT_PATH=${STAGING_DIR_TARGET},-DPLUGIN_DOWNLOADMANAGER=OFF -DLIB_PACKAGE=OFF,entservices-apis curl virtual/libpackage,entservices-apis curl virtual/libpackage"
+PACKAGECONFIG[ralfsupport] = "-DRALF_PACKAGE_SUPPORT=ON, -DRALF_PACKAGE_SUPPORT=OFF,ralf-utils jsoncpp, ralf-utils jsoncpp"
 # ----------------------------------------------------------------------------
 
 PACKAGEMANAGER_PLUGIN_ARGS         ?= " \
@@ -105,6 +108,7 @@ EXTRA_OECMAKE += " \
     -DSECAPI_LIB=sec_api \
     -DPLUGIN_NATIVEJS=ON \
     -DPLUGIN_NATIVEJS_CLIENTIDENTIFIER="${NATIVEJS_CLIENTIDENTIFIER}" \
+    -DPLUGIN_PREINSTALL_MANAGER_APP_PREINSTALL_DIRECTORY="${APP_PREINSTALL_DIRECTORY}" \
 "
 
 # TBD - set SECAPI_LIB to hw secapi once RDK-12682 changes are available
@@ -132,6 +136,7 @@ do_install:append() {
 
 FILES_SOLIBSDEV = ""
 FILES:${PN} += "${libdir}/wpeframework/plugins/*.so ${libdir}/*.so ${datadir}/WPEFramework/*"
+FILES:${PN} += "${@bb.utils.contains('DISTRO_FEATURES', 'enable_ralf', '${datadir}/*', '', d)}"
 
 INSANE_SKIP:${PN} += "libdir staticdev dev-so"
 INSANE_SKIP:${PN}-dbg += "libdir"
diff --git a/recipes-extended/packager-headers/packager-headers_git.bb b/recipes-extended/packager-headers/packager-headers_git.bb
index 2572edb3..9be9c71a 100644
--- a/recipes-extended/packager-headers/packager-headers_git.bb
+++ b/recipes-extended/packager-headers/packager-headers_git.bb
@@ -3,11 +3,11 @@ SUMMARY = "Package Headers"
 LICENSE = "Apache-2.0"
 LIC_FILES_CHKSUM = "file://LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57"
 
-PV = "1.0.0"
 PR = "r0"
 PACKAGE_ARCH = "${MIDDLEWARE_ARCH}"
 
-SRCREV = "1c2ea8e54b46e97c2cbcbfd1af348efea493060c"
+PV = "1.2.0"
+SRCREV = "92435d5ddb9541f25ee1887b84241d597aa37624"
 SRC_URI = "${CMF_GITHUB_ROOT}/eshelpers;${CMF_GITHUB_SRC_URI_SUFFIX};name=eshelpers"
 
 S = "${WORKDIR}/git"
diff --git a/recipes-extended/packager-lisa/packager-lisa_git.bb b/recipes-extended/packager-lisa/packager-lisa_git.bb
index 3c88b9e8..78875f73 100644
--- a/recipes-extended/packager-lisa/packager-lisa_git.bb
+++ b/recipes-extended/packager-lisa/packager-lisa_git.bb
@@ -4,18 +4,16 @@ SECTION = "rdk/libs"
 LICENSE = "Apache-2.0"
 LIC_FILES_CHKSUM = "file://LICENSE;md5=175792518e4ac015ab6696d16c4f607e"
 
-PV = "1.0.0"
+PV = "1.2.2"
 PR = "r0"
 PACKAGE_ARCH = "${MIDDLEWARE_ARCH}"
 
 PROVIDES = "virtual/libpackage"
 RPROVIDES:${PN} = "virtual/libpackage"
 
-DEPENDS = "sqlite3 boost libarchive"
 DEPENDS += "packager-headers"
-RDEPENDS_${PN} = " sqlite3 boost libarchive"
 
-SRCREV = "53fea63b4fb6e5491364781dcdfd91bcc47397f8"
+SRCREV = "79d57fa660c0772463ff497567a19a740002468e"
 SRC_URI = "${CMF_GITHUB_ROOT}/libpackage;${CMF_GITHUB_SRC_URI_SUFFIX};name=lisapack"
 SRCREV_FORMAT = "lisapack"
 
@@ -23,3 +21,13 @@ S = "${WORKDIR}/git"
 
 inherit cmake pkgconfig
 
+PACKAGECONFIG ?= " \
+    ${@bb.utils.contains('DISTRO_FEATURES', 'enable_ralf', 'ralfsupport', '', d)} \
+    ${@bb.utils.contains('DISTRO_FEATURES', 'DAC_SUPPORT', 'lisa', '', d)} \
+    "
+PACKAGECONFIG[ralfsupport]    = "-DENABLE_RALF_SUPPORT=ON -DDAC_APP_PATH=${DAC_APP_PATH}, -DENABLE_RALF_SUPPORT=OFF, ralf-utils jsoncpp, ralf-utils jsoncpp"
+PACKAGECONFIG[depcheck]    = "-DDISABLE_DEPENDENCY_CHECK=OFF, -DDISABLE_DEPENDENCY_CHECK=ON"
+PACKAGECONFIG[lisa]    = ",,sqlite3 boost libarchive,sqlite3 boost libarchive"
+
+DAC_APP_CERT_PATH ?= "/etc/rdk/certs"
+EXTRA_OECMAKE:append = " -DRDK_PACKAGE_CERT_PATH=${DAC_APP_CERT_PATH} -DDAC_APP_PATH=${DAC_APP_PATH}"
diff --git a/recipes-extended/sceneset/sceneset_git.bb b/recipes-extended/sceneset/sceneset_git.bb
index 5f62935e..838c4058 100644
--- a/recipes-extended/sceneset/sceneset_git.bb
+++ b/recipes-extended/sceneset/sceneset_git.bb
@@ -3,7 +3,7 @@ SUMMARY = "This recipe provides the sceneset component for RDK "
 LICENSE = "Apache-2.0"
 LIC_FILES_CHKSUM = "file://LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57"
 
-PV = "0.1.0"
+PV = "0.2.0"
 PR = "r0"
 PACKAGE_ARCH = "${MIDDLEWARE_ARCH}"
 
@@ -11,7 +11,7 @@ inherit cmake pkgconfig systemd
 
 DEPENDS += "wpeframework entservices-apis"
 
-SRCREV = "bd0bfe2c11788158b8025756d6b61f38304177a7"
+SRCREV = "18d540d6d0407506c71d4257743ebe8b73266997"
 SRC_URI = "${CMF_GITHUB_ROOT}/sceneset;${CMF_GITHUB_SRC_URI_SUFFIX};name=sceneset"
 SRCREV_FORMAT = "sceneset"
 
@@ -26,3 +26,4 @@ FILES:${PN} += "${bindir}/*"
 FILES:${PN} += "${systemd_unitdir}/system/*"
 
 SYSTEMD_SERVICE:${PN} = "sceneset.service"
+EXTRA_OECMAKE = "-DSCENESET_DEFAULT_APPNAME='${SCENESET_DEFAULT_APPNAME}'"
diff --git a/recipes-extended/wpe-framework/entservices-apis.bb b/recipes-extended/wpe-framework/entservices-apis.bb
index f2cdbe8a..d4c09741 100644
--- a/recipes-extended/wpe-framework/entservices-apis.bb
+++ b/recipes-extended/wpe-framework/entservices-apis.bb
@@ -1,7 +1,7 @@
 SUMMARY = "entservices-apis"
 LICENSE = "Apache-2.0"
 LIC_FILES_CHKSUM = "file://LICENSE;md5=d8927f3331d2b3e321b7dd1925166d25"
-PV = "2.12.0"
+PV = "3.2.0"
 PR = "r0"
 
 PACKAGE_ARCH = "${MIDDLEWARE_ARCH}"
@@ -17,7 +17,7 @@ SRC_URI += "file://RDKEMW-1007.patch"
 
 
 # Tag 2.12.0
-SRCREV_entservices-apis = "eae01243d0ced2ea066b4bd0b5c5f501a30565f6"
+SRCREV_entservices-apis = "bae4a2be47286463c11bdc51b526bad1e51f095a"
 
 S = "${WORKDIR}/git"
 TOOLCHAIN = "gcc"
