diff --git a/recipes-transit/app-cert-installer/app-cert-installer.bb b/recipes-transit/app-cert-installer/app-cert-installer.bb
new file mode 100644
index 0000000..fcd8df7
--- /dev/null
+++ b/recipes-transit/app-cert-installer/app-cert-installer.bb
@@ -0,0 +1,35 @@
+SUMMARY = "Install RDK certificates"
+DESCRIPTION = "Installs certificates into /etc/rdk/certs"
+LICENSE = "CLOSED"
+
+inherit allarch
+
+PV = "1.0"
+PR = "r0"
+PACKAGE_ARCH = "${MIDDLEWARE_ARCH}"
+
+PROVIDES += "virtual/app-sign-certs"
+RPROVIDES:${PN} += "virtual/app-sign-certs"
+
+S = "${WORKDIR}"
+
+SRC_URI = " \
+    file://com.rdkcentral.ralf.crt \
+    file://com.rdkcentral.ralf.key \
+    file://com.rdkcentral.ralf.p12 \
+"
+
+CERT_FILES = "${@' '.join([u.split('file://',1)[1].split(';',1)[0] for u in d.getVar('SRC_URI').split() if u.startswith('file://')])}"
+
+do_install() {
+    install -d ${D}${sysconfdir}/rdk/certs
+    for cert in ${CERT_FILES}; do
+        mode=0644
+        case "$cert" in
+            *.key|*.p12) mode=0600 ;;
+        esac
+        install -m ${mode} ${WORKDIR}/${cert} ${D}${sysconfdir}/rdk/certs/${cert}
+    done
+}
+
+FILES:${PN} += "${sysconfdir}/rdk/certs"
diff --git a/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.crt b/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.crt
new file mode 100755
index 0000000..8b13789
--- /dev/null
+++ b/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.crt
@@ -0,0 +1 @@
+
diff --git a/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.key b/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.key
new file mode 100755
index 0000000..8b13789
--- /dev/null
+++ b/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.key
@@ -0,0 +1 @@
+
diff --git a/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.p12 b/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.p12
new file mode 100755
index 0000000..8b13789
--- /dev/null
+++ b/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.p12
@@ -0,0 +1 @@
+
