diff --git a/recipes-transit/app-cert-installer/app-cert-installer.bb b/recipes-transit/app-cert-installer/app-cert-installer.bb
new file mode 100644
index 0000000..fcd8df7
--- /dev/null
+++ b/recipes-transit/app-cert-installer/app-cert-installer.bb
@@ -0,0 +1,35 @@
+SUMMARY = "Install RDK certificates"
+DESCRIPTION = "Installs certificates into /etc/rdk/certs"
+LICENSE = "CLOSED"
+
+inherit allarch
+
+PV = "1.0"
+PR = "r0"
+PACKAGE_ARCH = "${MIDDLEWARE_ARCH}"
+
+PROVIDES += "virtual/app-sign-certs"
+RPROVIDES:${PN} += "virtual/app-sign-certs"
+
+S = "${WORKDIR}"
+
+SRC_URI = " \
+    file://com.rdkcentral.ralf.crt \
+    file://com.rdkcentral.ralf.key \
+    file://com.rdkcentral.ralf.p12 \
+"
+
+CERT_FILES = "${@' '.join([u.split('file://',1)[1].split(';',1)[0] for u in d.getVar('SRC_URI').split() if u.startswith('file://')])}"
+
+do_install() {
+    install -d ${D}${sysconfdir}/rdk/certs
+    for cert in ${CERT_FILES}; do
+        mode=0644
+        case "$cert" in
+            *.key|*.p12) mode=0600 ;;
+        esac
+        install -m ${mode} ${WORKDIR}/${cert} ${D}${sysconfdir}/rdk/certs/${cert}
+    done
+}
+
+FILES:${PN} += "${sysconfdir}/rdk/certs"
diff --git a/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.crt b/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.crt
new file mode 100755
index 0000000..8b13789
--- /dev/null
+++ b/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.crt
@@ -0,0 +1 @@
+
diff --git a/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.key b/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.key
new file mode 100755
index 0000000..8b13789
--- /dev/null
+++ b/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.key
@@ -0,0 +1 @@
+
diff --git a/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.p12 b/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.p12
new file mode 100755
index 0000000..8b13789
--- /dev/null
+++ b/recipes-transit/app-cert-installer/app-cert-installer/com.rdkcentral.ralf.p12
@@ -0,0 +1 @@
+
diff --git a/recipes-transit/entservices-rdkwmgr/entservices-rdkwmgr_git.bb b/recipes-transit/entservices-rdkwmgr/entservices-rdkwmgr_git.bb
index 3120c32..d944d85 100644
--- a/recipes-transit/entservices-rdkwmgr/entservices-rdkwmgr_git.bb
+++ b/recipes-transit/entservices-rdkwmgr/entservices-rdkwmgr_git.bb
@@ -7,7 +7,7 @@ S = "${WORKDIR}/pbt"
 
 
 
-SRC_URI = "${RDK_ARTIFACTS_BASE_URL}/rdkm-rdke-prebuilts/entservices-rdkwmgr/lib32-entservices-rdkwmgr_${PV}.tar.gz;subdir=pbt"
+SRC_URI = "https://artifactory.rdkcentral.com/artifactory/rdkm-rdke-prebuilts/entservices-rdkwmgr/lib32-entservices-rdkwmgr_3.20.0.tar.gz;subdir=pbt"
 SRC_URI[sha256sum] = "4f098fb8d519b07e3fa31cedaf5bf599bae8bf2c5c9f33a7e1c89a01d0343be5"
 PACKAGE_ARCH = "${MIDDLEWARE_ARCH}"
 do_install () {
diff --git a/recipes-transit/rdkwindowmanager/rdkwindowmanager_git.bb b/recipes-transit/rdkwindowmanager/rdkwindowmanager_git.bb
index b132dba..26ce2a1 100644
--- a/recipes-transit/rdkwindowmanager/rdkwindowmanager_git.bb
+++ b/recipes-transit/rdkwindowmanager/rdkwindowmanager_git.bb
@@ -6,7 +6,7 @@ PV = "1.9.0"
 PR = "r0"
 S = "${WORKDIR}/pbt"
 
-SRC_URI = "${RDK_ARTIFACTS_BASE_URL}/rdkm-rdke-prebuilts/rdkwindowmanager/lib32-rdkwindowmanager_${PV}.tar.gz;subdir=pbt"
+SRC_URI = "https://artifactory.rdkcentral.com/artifactory/rdkm-rdke-prebuilts/rdkwindowmanager/lib32-rdkwindowmanager_1.9.0.tar.gz;subdir=pbt"
 SRC_URI[sha256sum] = "1452dd12137946a50889b2d8a123760a7bb19e36289a9ca6ddd33a74321f2d70"
 
 do_install() {
